src/Sulu/Component/Content/Tests/Functional/Rlp/Strategy/TreeStrategyTest.php

```diff
@@ -0,0 +1,301 @@
+        $this->documentManager->flush();
+
+        $news2 = $this->createDocument($news, 'news-2', '/news/news-2');
+        $this->documentManager->persist($news2, 'de');
+        $this->documentManager->flush();
+
+        $news2sub1 = $this->createDocument($news2, 'news-2', '/news/news-2/sub-1');
+        $this->documentManager->persist($news2sub1, 'de');
+        $this->documentManager->flush();
+
+        $news2sub2 = $this->createDocument($news2, 'news-2', '/news/news-2/sub-2');
+        $this->documentManager->persist($news2sub2, 'de');
+        $this->documentManager->flush();
+
+        // move route
+        $news = $this->documentManager->find($news->getUuid(), 'de');
+        $news->setResourceSegment('/test');
+        $this->rlpStrategy->save($news, null);
+        $this->session->save();
+        $this->session->refresh(false);
+
+        // check exist new routes
+        $this->assertEquals(
+            $news->getUuid(),
+            $this->rlpStrategy->loadByResourceLocator('/test', 'sulu_io', 'de')
+        );
+        $this->assertEquals(
+            $news1->getUuid(),
+            $this->rlpStrategy->loadByResourceLocator('/test/news-1', 'sulu_io', 'de')
+        );
+        $this->assertEquals(
+            $news1sub1->getUuid(),
+            $this->rlpStrategy->loadByResourceLocator('/test/news-1/sub-1', 'sulu_io', 'de')
+        );
+        $this->assertEquals(
+            $news1sub2->getUuid(),
+            $this->rlpStrategy->loadByResourceLocator('/test/news-1/sub-2', 'sulu_io', 'de')
+        );
+
+        $this->assertEquals(
+            $news2->getUuid(),
+            $this->rlpStrategy->loadByResourceLocator('/test/news-2', 'sulu_io', 'de')
+        );
+        $this->assertEquals(
+            $news2sub1->getUuid(),
+            $this->rlpStrategy->loadByResourceLocator('/test/news-2/sub-1', 'sulu_io', 'de')
+        );
+        $this->assertEquals(
+            $news2sub2->getUuid(),
+            $this->rlpStrategy->loadByResourceLocator('/test/news-2/sub-2', 'sulu_io', 'de')
+        );
+
+        // check history
+        $this->assertEquals('/test', $this->getRlForHistory('/news'));
+        $this->assertEquals('/test/news-1', $this->getRlForHistory('/news/news-1'));
+        $this->assertEquals('/test/news-1/sub-1', $this->getRlForHistory('/news/news-1/sub-1'));
+        $this->assertEquals('/test/news-1/sub-2', $this->getRlForHistory('/news/news-1/sub-2'));
+
+        $this->assertEquals('/test/news-2', $this->getRlForHistory('/news/news-2'));
+        $this->assertEquals('/test/news-2/sub-1', $this->getRlForHistory('/news/news-2/sub-1'));
+        $this->assertEquals('/test/news-2/sub-2', $this->getRlForHistory('/news/news-2/sub-2'));
+    }
+
+    public function testRestore()
+    {
+        $rootNode = $this->session->getNode('/cmf/sulu_io/routes/de');
+        $parentDocument = $this->documentManager->find('/cmf/sulu_io/contents');
+
+        sleep(1);
```

Comment:
TBH, I just copied the stuff over, and don't know exactly why it is necessary. This specific line could be deleted, and it worked. But one of the sleeps here is necessary. I would not introduce a separate method for that, because a required sleep might have different causes. So i would say a comment is fine, but as said, I don't really know why the left one is still needed, so I can't really put an explanation there.